Civi/Api4/Action/MessageTemplate/GetContent.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+
+namespace Civi\Api4\Action\MessageTemplate;
+
+use Civi\Api4\Generic\Result;
+use Civi\Token\TokenProcessor;
+
+/**
+ * Class GetContent.
+ *
+ * Get the content of an email for the given template text, rendering tokens.
+ *
+ * @method int setMessageTemplateId(int $messageTemplateID) Set Message Template ID.
+ * @method int getMessageTemplateId(int $messageTemplateID) Get Message Template ID.
+ * @method string setMessageSubject(string $messageSubject) Set Message Subject
+ * @method string getMessageSubject(string $messageSubject) Get Message Subject
+ * @method string setMessageHtml(string $messageHtml) Set Message Html
+ * @method string getMessageHtml string $messageHtml) Get Message Html
+ * @method string setMessageText(string $messageHtml) Set Message Text
+ * @method string getMessageText string $messageHtml) Get Message Text
+ * @method string getAdHocStringsToParse(array $stringToParse) Get array of adhoc strings to parse.
+ * @method string setAdHocStringsToParse(array $stringToParse) Set array of adhoc strings to parse.
+ * @method array setEntities(string $entities) Set entity/entities.
+ * @method array getEntities(string $entities) Get entity/entities.
+ * @method array setEntityIds(array $entityIds) Set entity IDs
+ * @method array getEntityIds(array $entityIds) Get entity IDs
+ */
+class GetContent extends \Civi\Api4\Generic\AbstractAction {
+
+  /**
+   * ID of message template.
+   *
+   * It is necessary to pass this or at least one of message_subject, message_text or message_html.
+   *
+   * @var int
+   */
+  protected $messageTemplateId;
+
+  /**
+   * Ad hoc html strings to parse.
+   *
+   * Array of adhoc strings arrays to pass e.g
+   *  [
+   *    ['string' => 'Dear {contact.first_name}', 'format' => 'text/html', 'key' => 'greeting'],
+   *    ['string' => 'Also known as {contact.first_name}', 'format' => 'text/plain', 'key' => 'nick_name'],
+   * ]
+   *
+   * If no provided the key will default to 'string' and the format will default to 'text'
+   *
+   * @var array
```

Comment:
Use the @requiredIf annotation like:

    @requiredIf empty($messageTemplateId)

Reply:
Nevermind that doesn't work as an annotation.